src/apikey/routers/manage.js

```diff
@@ -0,0 +1,355 @@
+          return res.status(400).json({ message: `argument "access" [${e}] doesn't exist` });
+        }
+      });
+    }
+  }
+
+  if (config.allowed) {
+    if (typeof config.allowed !== 'boolean') {
+      return res.status(400).json({ message: `argument "allowed" [${config.allowed}] is in wrong format` });
+    }
+  }
+
+  if (config.attributes) {
+    if (typeof config.attributes !== 'string') {
+      return res.status(400).json({ message: `argument "attributes" [${config.attributes}] is in wrong format` });
+    }
+
+    const attrs = config.attributes.split(',');
+    attrs.forEach((attr) => {
+      if (!unpaywallAttrs.includes(attr)) {
+        return res.status(400).json({ message: `argument "attributes" [${attr}] doesn't exist` });
+      }
+    });
+  }
+
+  let key;
+  try {
+    key = await redisClient.get(apikey);
+  } catch (err) {
+    logger.error(`Cannot get ${apikey} on redis`);
+    logger.error(err);
+    return res.status(500).json({ message: 'Internal server error' });
+  }
+
+  if (!key) {
+    return res.status(404).json({ message: `[${apikey}] apikey doesn't exist` });
+  }
+
+  try {
+    await updateAuth(apikey, config.name, config.access, config.attributes, config.allowed);
+  } catch (err) {
+    return next(err);
+  }
+
+  let configApiKey = await redisClient.get(apikey);
+  configApiKey = JSON.parse(configApiKey);
+
+  return res.status(200).json({ apikey, config: configApiKey });
+});
+
+/**
+ * delete apikey
+ */
+router.delete('/delete', checkAuth, async (req, res, next) => {
+  const { apikey } = req.body;
+
+  if (!apikey) {
+    return res.status(400).json({ message: 'apikey expected' });
+  }
+
+  let key;
+  try {
+    key = await redisClient.get(apikey);
+  } catch (err) {
+    logger.error(`Cannot get ${apikey} on redis`);
+    logger.error(err);
+    return res.status(500).json({ message: 'Internal server error' });
+  }
+
+  if (!key) {
+    return res.status(404).json({ message: `[${apikey}] apikey doesn't exist` });
+  }
+
+  try {
+    await deleteAuth(apikey);
+  } catch (err) {
+    return next(err);
+  }
+
+  return res.status(204).json({});
+});
+
+/**
+ * delete all apikey
+ */
+router.delete('/all', checkAuth, async (req, res, next) => {
+  try {
+    await redisClient.flushall();
+  } catch (err) {
+    return next(err);
+  }
+  return res.status(204).json();
+});
+
+/**
+ * load apikey
+ */
+router.post('/load', checkAuth, async (req, res, next) => {
+  const { dev } = req.query;
+  const { keys } = req.body;
+  if (dev) {
+    try {
+      await load();
+    } catch (err) {
+      return next(err);
+    }
+    return res.status(204).json();
+  }
+
+  const apiKeysJSON = Object.keys(keys);
+
+  apiKeysJSON.forEach(async (key) => {
+    try {
+      await redisClient.set(key, `${JSON.stringify(keys[key])}`);
+    } catch (err) {
+      logger.error(`Cannot load ${key} with ${JSON.stringify(keys[key])} on redis`);
+      logger.error(err);
+    }
+  });
```

Comment:
Same as the `load()` function in `lib/redis.js`, you should either use `Promise.all()` or a `for loop` to be sure that the keys are actually loaded when the server responds.